enhancements/bootimages.md

```diff
@@ -0,0 +1,163 @@
+
+#### Exposing a general purpose "OS build" tool
+
+This proposal includes shipping a subset of https://github.com/coreos/coreos-assembler/ - but that's not the same as supporting custom "CoreOS-style" builds for non-OpenShift use cases.
+
+## Proposal phase 1: bootimage.json in release image
+
+First, the RHCOS build process is changed to inject the current coreos-assembler `meta.json` output for the build into `machine-os-content`.  This aims to move the "source of truth" for cluster bootimages into the release image.  Nothing would use this to start - however, as soon as we switch to a machine API provisioned control plane, having that process consume this data would be a natural next step.
+
+In fact, we could aim to switch to having workers use the `bootimage.json` from the release payload immediately after it lands.  A downside is this would open up potential for drift between the bootstrap+controlplane and workers.
+
+## Proposal phase 2: oc adm release generate-bootimage
+
+The implementation of this is basically shipping a subset of [coreos-assembler](https://github.com/coreos/coreos-assembler) as part of the OpenShift release payload, and teaching `oc` how to invoke `podman` to run it.
+
+The `generate-bootimage` implementation would download the `machine-os-bootimage-generator` container image along with the existing `machine-os-content` container image (OSTree repository), and effectively run the [create_disk.sh](https://github.com/coreos/coreos-assembler/blob/30fbac4e176c7936362efbd647c8199d927e593c/src/create_disk.sh) process or [buildextend-installer](https://github.com/coreos/coreos-assembler/blob/30fbac4e176c7936362efbd647c8199d927e593c/src/cmd-buildextend-installer) for live media, etc.
+
+This should have three possible runtime choices:
+
+- Run via `--privileged` and use loopback mounts (avoids any virtualization requirements)
+- Run via `--device /dev/kvm` (as coreos-assembler is optimized for today)
+- Run with `--env COSA_NO_KVM=1` to run in environments without KVM
+
+It would also be theoretically possible to support using e.g. an on-demand provisioned EBS volume, but this would impose a burden on the CoreOS team for another build path.
+
+This `oc` enhancement would currently *not* run directly on platforms that don't have `podman` (i.e. OS X/Windows).  But we could likely tweak things so that it could use e.g. the Docker API and Docker Desktop for Windows or equivalent.
+
+## Proposal phase 3: machine-api-bootimage-updater
+
+Add a new component to https://github.com/openshift/machine-api-operator which runs `machine-os-bootimage-generator`, uploads the resulting images to the IaaS layer (AMI, OpenStack Glance, etc.) and updates the machinesets for the cluster.
```

Comment:
Are we moving away from public AMIs?

There was some talk about possibly changing machine-controller to an image reference rather than updating each machineset.  Either of these cases has the question of mixed image support.  Do we plan on doing that in the future, or do we do it now?

Reply:
https://github.com/openshift/installer/issues/2906 suggests moving away from AMIs, yep.